shared/src/main/scala/kinesis4cats/consumer/lambda/KinesisStreamEvent.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2023-2023 etspaceman
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kinesis4cats.consumer
+package lambda
+
+import scala.util.Try
+
+import java.time.Instant
+
+import scodec.bits.ByteVector
+
+import kinesis4cats.models.EncryptionType
+
+final case class KinesisStreamRecordPayload(
```

Comment:
One thing I am wondering is if I should create a separate module for these classes. Main reason I suggest that is that I can't include the circe decoders in the companion objects in `shared` without bringing circe dependencies into that module. I could create a separate module that introduces `circe` as well. For now, i have the circe decoders in the `shared-circe` module.

Reply:
you should consider this pattern: https://blog.7mind.io/no-more-orphans.html

Reply:
Has the `Optional` thing seen real production usage? I remember Gavin having some concerns about it, but not remember exactly which ones.

Reply:
it works for Caliban, I don't know much about its flaws ;)

Reply:
I think I'm going to move this to its own module. Seems like the best thing here if we need to evolve it with Feral's needs.